tests/models/test_url.py

```diff
@@ -287,3 +287,37 @@ def test_url_with_url_encoded_path():
     assert url.path == "/path to somewhere"
     assert url.query == b""
     assert url.raw_path == b"/path%20to%20somewhere"
+
+
+def test_ipv6_url():
+    url = httpx.URL("http://[::ffff:192.168.0.1]:5678/")
+
+    assert url.host == "::ffff:192.168.0.1"
+    assert url.netloc == "[::ffff:192.168.0.1]:5678"
+
+
+@pytest.mark.parametrize(
+    "url_str",
+    [
+        "http://127.0.0.1:1234",
+        "http://example.com:1234",
+        "http://[::ffff:192.168.0.2]:1234",
```

Comment:
Change this host as the above
```suggestion
        "http://[::ffff:127.0.0.1]:1234",
```